train/tictactoe.py

```diff
@@ -40,41 +41,65 @@ def load_data_entry(self, path):
 
         return (planes, probs, winner)
 
+    def _get_input_shape(self):
+        cpu = True
+        shape_cpu = (self.BOARD_SIZE, self.BOARD_SIZE, self.PLANES_NUM)
+        shape_gpu = (self.PLANES_NUM, self.BOARD_SIZE, self.BOARD_SIZE)
+        return shape_cpu if cpu else shape_gpu
+
     def _create_model_simple_two_headed(self):
-        input_layer = Input(
-            shape=(self.PLANES_NUM, self.BOARD_SIZE, self.BOARD_SIZE),
+        inputs = Input(
+            shape=self._get_input_shape(),
             name="in_position")
-        x = Dense(units=9, activation="relu")(input_layer)
-        output_scalar_layer = Dense(
+        x = Dense(units=9, activation="relu")(inputs)
+        head_val = Dense(
             units=1, activation="tanh", name="out_value")(x)
-        output_probs_layer = Dense(
+        head_probs = Dense(
             units=self.MOVE_NUM, activation="sigmoid", name="out_probs")(x)
 
-        model = Model(inputs=input_layer, outputs=[
-            output_scalar_layer, output_probs_layer])
+        model = Model(inputs=inputs, outputs=[head_val, head_probs])
 
         opt = optimizers.Adam(learning_rate=_LEARNING_RATE)
         model.compile(
             optimizer=opt,
-            loss={
-                'out_value': tf.keras.losses.MeanSquaredError(),
-                'out_probs': net_utils.loss_cross_entropy
-            },
+            loss={'out_value': tf.keras.losses.MeanSquaredError(),
+                  'out_probs': net_utils.loss_cross_entropy},
             metrics={'out_value': tf.keras.metrics.MeanSquaredError(),
                      'out_probs': tf.keras.metrics.KLDivergence()})
         return model
 
-    def create_model(self, net_type: str) -> keras.Model:
+    def _create_model_convnetv1(self, cfg):
+        inputs = Input(
+            shape=self._get_input_shape(),
+            name="in_position")
+        outputs = net_utils.create_two_headed_net(
+            inputs,
+            residual_filter_num=cfg["residual_filter_num"],
+            residual_block_num=cfg["residual_block_num"],
+            moves_num=self.MOVE_NUM)
+        model = Model(inputs=inputs, outputs=outputs)
+
+        opt = optimizers.Adam(learning_rate=_LEARNING_RATE)
+        model.compile(
+            optimizer=opt,
+            loss={'out_value': tf.keras.losses.MeanSquaredError(),
+                  'out_probs': net_utils.loss_cross_entropy},
+            metrics={'out_value': tf.keras.metrics.RootMeanSquaredError(),
```

Comment:
This is taken from the Alpha Zero paper? 
I will have to read a bit of documentation to confirm you on this

Reply:
![image](https://user-images.githubusercontent.com/15381794/191585587-97b43ffe-5663-4d01-8fd8-4c3e9e24a49a.png)